### PR TITLE
Rewriting the README

### PR DESCRIPTION
**Pull request by irace**
https://github.com/tumblr/XExtensionItem/issues/32

Rewriting the README to incorporate all of the API changes. **I’d absolutely _love_ your feedback on how we can make this more comprehensive yet as approachable and clear as possible.**

Some links may be broken until [the Carthage PR](https://github.com/tumblr/XExtensionItem/pull/47) is merged; I’ll handle that once it’s in.

There’s also a link to a [wiki page](https://github.com/tumblr/XExtensionItem/wiki/Guide-to-adding-custom-parameters) which I’ve yet to write. This’ll happen tomorrow.

@arix @mattbischoff @prendio2 @padraigk

### CODE REVIEW

**Comment by prendio2** [comment]
This looks great. I think I may have missed some of the conversation about API changes because the new API is even simpler and more intuitive than I remembered. Until I saw the new README I wasn't aware additional attachments were able to be created using standard objects like strings and images. 

It might be nice to show an example of the item providing blocks in use, the point that they exist is perhaps lost on a quick scan without an example.

Using Instapaper as an example of an app that doesn't accept images, even hypothetically, may be misleading once the next update comes out which addresses that. (Of course apps having to pretend they accept images when there's nothing they can do with them is symptomatic of the whole issue)

Tumblr should be listed in the "Apps that use XExtensionItem" section… We use it in Unread but not sure that's particularly pertinent at the moment unless we start add additional custom parameters for certain extensions in the future.

**Comment by mattbischoff** [comment]
The Copyright should be 2015.

**Comment by mattbischoff** [comment]
> @{ @"tumblr-custom-url": @"/post/123/best-post-ever" }

Wouldn’t we want them to use public constants for the keys?

**Comment by irace** [comment]
> Wouldn’t we want them to use public constants for the keys?

That's removed in this PR, no?

**Comment by mattbischoff** [comment]
Apparently I’m going colorblind.

File: README.md [hunk 1]
```diff
@@ -1,127 +1,159 @@
-![XExtensionItem](https://github.com/tumblr/XExtensionItem/blob/master/Assets/logo.png)
+<p align="center"><img src="https://github.com/tumblr/XExtensionItem/blob/master/Assets/logo.png"></p>
 
-# XExtensionItem
+<h1 align="center">XExtensionItem</p>
```
Comment (mattbischoff) [inline README.md]: > <h1 align="center">XExtensionItem</p>

Did you mean <h1 align="center">XExtensionItem</h1>?

File: README.md [hunk 2]
```diff
@@ -1,127 +1,159 @@
...
+- [Examples](#examples)
+- [Apps that use XExtensionItem](#apps-that-use-xextensionitem)
+- [Contributing](#contributing)
+- [Contact](#contact)
+- [License](#license)
 
 ## Why?
 
-Currently, share extensions have an [unfortunate limitation](https://github.com/tumblr/ios-extension-issues/issues/5) which causes only applications that explicitly accept *all* provided activity item types to show up in a `UIActivityViewController`. This makes it difficult to share multiple pieces of data without causing extensions that aren’t as flexible with their allowed inputs to not show up at all. Consider the following example:
+Currently, iOS has an [unfortunate limitation](https://github.com/tumblr/ios-extension-issues/issues/5) which causes only share extensions that explicitly accept *all* provided activity item types to show up in a `UIActivityViewController`. This makes it difficult for an application to share multiple pieces of data without causing extensions that aren’t as flexible with their allowed inputs to not show up at all. Consider the following example:
```
Comment (mattbischoff) [inline README.md]: > share extensions

Does this affect action extensions also?

File: README.md [hunk 3]
```diff
@@ -1,127 +1,159 @@
...
 
-#### Applications
+* A title
+* Attributed content text (can also be specific on a per-activity type basis)
+* A thumbnail image
+* Tags
+* A source URL
+* Referrer information
+    * App Name
+    * App store IDs (iTunes and Google Play)
```
Comment (mattbischoff) [inline README.md]: App store → App Store

File: README.md [hunk 4]
```diff
@@ -148,6 +189,10 @@ Please see [CONTRIBUTING.md](https://github.com/tumblr/XExtensionItem/blob/maste
 
 [Bryan Irace](mailto:bryan@tumblr.com)
 
+### Thank you
+
+Many thanks to [Ari Weinstein](https://github.com/arix) for his contributions towards shaping XExtensionItem’s API, as well as [Matt Bischoff](https://github.com/mattbischoff), [Oisín Prendiville](https://github.com/prendio2), and [Padraig Kennedy](https://github.com/padraigk) for their invaluable feedback.
```
Comment (mattbischoff) [inline README.md]: <3

**Comment by mattbischoff** [comment]
Looking really good, but I feel like I might already know too much to read this with fresh eyes. Maybe @khanlou, @cdzombak, @grantjbutler, or @bcapps would have more useful first-read feedback. https://github.com/tumblr/XExtensionItem/blob/feature/readme/README.md

**Comment by irace** [comment]
> This looks great. I think I may have missed some of the conversation about API changes because the new API is even simpler and more intuitive than I remembered. Until I saw the new README I wasn't aware additional attachments were able to be created using standard objects like strings and images.

Yes, @AriX made these changes. They’re great!

> It might be nice to show an example of the item providing blocks in use, the point that they exist is perhaps lost on a quick scan without an example.

Sure thing, I’ll add one.

> Using Instapaper as an example of an app that doesn't accept images, even hypothetically, may be misleading once the next update comes out which addresses that. (Of course apps having to pretend they accept images when there's nothing they can do with them is symptomatic of the whole issue).

Good call, I’ll choose a different example.

> Tumblr should be listed in the "Apps that use XExtensionItem" section… We use it in Unread but not sure that's particularly pertinent at the moment unless we start add additional custom parameters for certain extensions in the future.

Yep, totally. I’ll make separate sections for apps that use it vs. extensions that use it. I’d like to list Unread even though it doesn’t have any custom parameter support.

**Comment by irace** [comment]
I’m going to merge this, but I’d still love any feedback that anyone has!

File: README.md (hunk 2 above)
Comment (irace) [inline README.md]: ¯\\_(ツ)_/¯

File: README.md (hunk 3 above)
Comment (irace) [inline README.md]: It’s not a proper noun, I’m referring to both iTunes _and_ Google Play.

**Comment by AriX** [comment]
README looks great. The thought I had when reading through it is that (especially in the "Why?" section) you mostly talk about the usefulness of XExtensionItemSource, but not about the greater aspirations of allowing apps to pass richer metadata to each other and custom parameters and such. Just wanted to point that out!

**Comment by irace** [comment]
Thanks @AriX – I’ve elaborated a bit to hopefully address your feedback: https://github.com/tumblr/XExtensionItem/commit/f0307c2ead160e82f70d40b4c5c4077de0667425

**Comment by AriX** [comment]
Sweet!

**Pull request merged**